src/controllers/bundles.js

```diff
@@ -0,0 +1,192 @@
+ */
+export async function getDataChunks(url, domainkey, startdate, enddate) {
+  if (!url || !domainkey) {
+    throw new Error('Both url and domainkey are required.');
+  }
+  let start;
+  let end;
+
+  if (!startdate || !enddate) {
+    start = new Date().toISOString().slice(0, 10).replace(/-/g, '/');
+    end = new Date().toISOString().slice(0, 10).replace(/-/g, '/');
+  } else {
+    start = new Date(startdate);
+    end = new Date(enddate);
+  }
+
+  const dateList = [];
+
+  for (let d = new Date(start); d <= end; d.setDate(d.getDate() + 1)) {
+    dateList.push(d.toISOString().slice(0, 10).replace(/-/g, '/'));
+  }
+
+  const r = (await Promise.all(dateList.map((date) => loadBundles(url, date, domainkey)))).flat();
+
+  return r;
+}
+
+/**
+ * Get the relevant statistic for a DataChunk.
+ *
+ * @param {string} dataChunks - Rum Bundle Datachunks.
+ * @param {string} aggregation - Statistic Agent Needs (pageviews, lcp, inp, etc.).
+ * @returns {Promise<object>} - The relevant statistic.
+ */
+export async function getStatistic(url, dataChunks, aggregation) {
+  if (!dataChunks || !aggregation) {
+    throw new Error('dataChunks, aggregation, and statistic are required.');
+  }
+
+  let aggHandler;
+
+  if (aggregation === 'pageviews') {
+    aggHandler = series.pageViews;
+  } else if (aggregation === 'visits') {
+    aggHandler = series.visits;
+  } else if (aggregation === 'bounces') {
+    aggHandler = series.bounces;
+  } else if (aggregation === 'organic') {
+    aggHandler = series.organic;
+  } else if (aggregation === 'earned') {
+    aggHandler = series.earned;
+  } else if (aggregation === 'lcp') {
+    aggHandler = series.lcp;
+  } else if (aggregation === 'cls') {
+    aggHandler = series.cls;
+  } else if (aggregation === 'inp') {
+    aggHandler = series.inp;
+  } else if (aggregation === 'ttfb') {
+    aggHandler = series.ttfb;
+  } else if (aggregation === 'engagement') {
+    aggHandler = series.engagement;
+  } else {
+    throw new Error(`Unsupported aggregation: ${aggregation}`);
+  }
+
+  dataChunks.forEach((chunk) => {
+    // eslint-disable-next-line no-param-reassign
+    chunk.rumBundles = chunk.rumBundles.map(utils.addCalculatedProps);
+  });
+
+  const d = new DataChunks();
+  d.load(dataChunks);
+  d.filteredIn = filterBundlesByUrl(url, d.bundles);
+  d.addSeries(aggregation, aggHandler);
+  d.group((b) => b.url);
+
+  const resultList = [];
+  for (const [urlL, metrics] of Object.entries(d.aggregates)) {
+    const metric = metrics[aggregation];
+    if (metric && metric.count > 0) {
+      resultList.push({
+        urlL,
+        count: metric.count,
+        sum: metric.sum,
+        mean: metric.mean,
+        p50: metric.percentile(50),
+        p75: metric.percentile(75),
+      });
+    }
+  }
+  return resultList;
+}
+
+/**
+ * Bundles controller.
+ * @param {object} ctx - Context object.
+ * @returns {object} Bundles controller.
+ * @constructor
+ */
+function BundlesController(ctx) {
```

Comment:
i don't understand why you're adding an HTTP controller. this is only needed if you want to expose your data via our HTTP API, in which case first an OpenAPI spec would have to be submitted and the controller be added to our HTTP routes. MCP alone does not require an HTTP controller. For simplicity reasons we wired existing HTTP controller actions that we also wanted to be available as tool to the controllers, this doesn't mean all new tools/resources have to implement a controller.